editor/plugins/animation_state_machine_editor.h

```diff
@@ -64,7 +69,36 @@ class AnimationNodeStateMachineEditor : public AnimationTreeNodeEditorPlugin {
 
 	OptionButton *play_mode = nullptr;
 
+	HSplitContainer *hsplit = nullptr;
+	MarginContainer *sidepanel_rect = nullptr;
+
 	PanelContainer *panel = nullptr;
+	PanelContainer *sidepanel = nullptr;
+
+	VBoxContainer *sp_vertical_partition = nullptr;
+
+	HBoxContainer *sp_title_container = nullptr;
+	Label *sidepanel_title = nullptr;
+	Label *sidepanel_title_2 = nullptr;
+	HSeparator *h_sep = nullptr;
+	HBoxContainer *visibility_reset_container = nullptr;
+	Button *show_all_node_group_button = nullptr;
+	Button *hide_all_node_group_button = nullptr;
+
+	HBoxContainer *node_group_hor_containers[12] = { nullptr };
+	HBoxContainer *node_group_containers[12] = { nullptr };
+	Button *show_node_group_buttons[12] = { nullptr };
+	Button *hide_node_group_buttons[12] = { nullptr };
+	LineEdit *node_group_name_line[12] = { nullptr };
+	CheckBox *node_group_select_checkboxes[12] = { nullptr };
+	Label *node_group_labels[12] = { nullptr };
```

Comment:
```suggestion
	HBoxContainer *node_group_hor_containers[AnimationNodeStateMachine::STATE_MACHINE_GROUPS_LIMIT] = { nullptr };
	HBoxContainer *node_group_containers[AnimationNodeStateMachine::STATE_MACHINE_GROUPS_LIMIT] = { nullptr };
	Button *show_node_group_buttons[AnimationNodeStateMachine::STATE_MACHINE_GROUPS_LIMIT] = { nullptr };
	Button *hide_node_group_buttons[AnimationNodeStateMachine::STATE_MACHINE_GROUPS_LIMIT] = { nullptr };
	LineEdit *node_group_name_line[AnimationNodeStateMachine::STATE_MACHINE_GROUPS_LIMIT] = { nullptr };
	CheckBox *node_group_select_checkboxes[AnimationNodeStateMachine::STATE_MACHINE_GROUPS_LIMIT] = { nullptr };
	Label *node_group_labels[AnimationNodeStateMachine::STATE_MACHINE_GROUPS_LIMIT] = { nullptr };
```
It also might be replace-able if you predefine or include AnimationNodeStateMachine.